stl/inc/functional

```diff
@@ -2204,7 +2204,9 @@ _CONSTEXPR20 pair<_FwdItHaystack, _FwdItHaystack> _Search_pair_unchecked(
         }

```

Comment:
Can we never use the random-access iterator path above with iterators that model `random_access_iterator` but whose iterator category is not `random_access_iterator_tag`?

Reply:
I didn't want to make changes to behavior that weren't specified explicitly. I can do it differently if you think that's best, though.